agent/checks/grpc.go

```diff
@@ -32,38 +35,39 @@ func NewGrpcHealthProbe(target string, timeout time.Duration, tlsConfig *tls.Con
 		request.Service = serverAndService[1]
 	}
 
-	var dialOptions = []grpc.DialOption{}
-
+	var dialOptions []grpc.DialOption
 	if tlsConfig != nil {
 		dialOptions = append(dialOptions, grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)))
 	} else {
 		dialOptions = append(dialOptions, grpc.WithInsecure())
 	}
 
+	serverWithScheme := fmt.Sprintf("%s:///%s", resolver.GetDefaultScheme(), serverAndService[0])
+	connection, err := grpc.DialContext(context.Background(), serverWithScheme, dialOptions...)
+	if err != nil {
+		// dial may return an error only when opts are invalid or grpc.WithBlock is used
+		panic(err)
```

Comment:
Thanks for the comment here. Currently I don't believe we are using `grpc.WithBlock` in our code, but it's a good call out for future work.

Some feedback around L#49 - Is there a particular reason why a panic is being used here? 

Another way to handle the failure case is to log that there is an error, and attempt to retry the connection. The structs themselves don't return an error, so bubbling up the failure and wrapping it may be more effective.